storage/remote/otlptranslator/prometheusremotewrite/helper.go

```diff
@@ -205,13 +208,22 @@ func createAttributes(resource pcommon.Resource, attributes pcommon.Map, scope s
 
 	if settings.EnableTypeAndUnitLabels {
 		unitNamer := otlptranslator.UnitNamer{UTF8Allowed: settings.AllowUTF8}
+
 		if metadata.Type != prompb.MetricMetadata_UNKNOWN {
 			l["__type__"] = strings.ToLower(metadata.Type.String())
 		}
 		if metadata.Unit != "" {
 			l["__unit__"] = unitNamer.Build(metadata.Unit)
 		}
 	}
+	if settings.AllowDeltaTemporality && hasTemporality {
+		switch temporality {
+		case pmetric.AggregationTemporalityCumulative:
+			l["__temporality__"] = "cumulative"
+		case pmetric.AggregationTemporalityDelta:
+			l["__temporality__"] = "delta"
```

Comment:
I think you should add a `default` case, so you can fail when an unrecognized `AggregationTemporalityCumulative` is encountered. Otherwise, debugging would be a nightmare in case an invalid value is passed in.